src/core/main.qml

```diff
@@ -47,121 +47,152 @@ Window {
+            
             onTriggered: {
                 DownloadManager.voicesRegistered.disconnect(playWelcome);
             }
-
+            
             function playWelcome() {
                 audioVoices.append(ApplicationInfo.getAudioFilePath("voices-$CA/$LOCALE/misc/welcome.$CA"));
             }
         }
-
+        
         Component.onCompleted: {
-            if(ApplicationSettings.isAudioEffectsEnabled)
-                append(ApplicationInfo.getAudioFilePath("qrc:/gcompris/src/core/resource/intro.$CA"))
+            if(!ApplicationSettings.isAudioEffectsEnabled && isBackgroundMusicAllowed) {
+                 Math.floor(Math.random()*rccBackgroundMusic.length);
+                backgroundMusic.append(ApplicationInfo.getAudioFilePath("backgroundMusic/" + rccBackgroundMusic[selectedMusicIndex]));
 
-            if (DownloadManager.areVoicesRegistered())
-                delayedWelcomeTimer.playWelcome();
-            else {
-                DownloadManager.voicesRegistered.connect(
-                        delayedWelcomeTimer.playWelcome);
-                delayedWelcomeTimer.start();
+            }
+
+           else  if(ApplicationSettings.isAudioEffectsEnabled)
+            {
+                append(ApplicationInfo.getAudioFilePath("qrc:/gcompris/src/core/resource/intro.$CA"))
+                if (DownloadManager.areVoicesRegistered()){
+                    delayedWelcomeTimer.playWelcome();
+                    delayedbackgroundMusic.start();
+                }
+                else {
+                    DownloadManager.voicesRegistered.connect(
+                                delayedWelcomeTimer.playWelcome);
+                    delayedWelcomeTimer.start();
+                    delayedbackgroundMusic.start();
+                }
             }
         }
     }
-
+    
+    Timer {
+        id: delayedbackgroundMusic
+        interval: 20000
+        repeat: false
+        
+        onTriggered: {
+            print(rccBackgroundMusic)
+            selectedMusicIndex = Math.floor(Math.random()*rccBackgroundMusic.length);
+            backgroundMusic.append(ApplicationInfo.getAudioFilePath("backgroundMusic/"+rccBackgroundMusic[selectedMusicIndex]));
+        }
+    }
+    
     GCAudio {
         id: audioEffects
         muted: !ApplicationSettings.isAudioEffectsEnabled
     }
+    
+    GCAudio {
+        id: backgroundMusic
+        muted: !ApplicationSettings.isBackgroundMusicEnabled
+        
+        onDone: {
+            selectedMusicIndex = Math.floor(Math.random()*rccBackgroundMusic.length);
+            backgroundMusic.append(ApplicationInfo.getAudioFilePath("backgroundMusic/" + rccBackgroundMusic[selectedMusicIndex]));
+        }
+    }
 
     function playIntroVoice(name) {
         name = name.split("/")[0]
         audioVoices.append(ApplicationInfo.getAudioFilePath("voices-$CA/$LOCALE/intro/" + name + ".$CA"))
     }
-
+    
     function checkWordset() {
-        var wordset = ApplicationSettings.wordset
```

Comment:
Actually I suggested on IRC to do the initial updating/downloading/registering of backgroundMusic in this function equivalently to wordset.